src/Application/Job/DataDumpJob.php

```diff
@@ -0,0 +1,148 @@
+            throw new ServiceNotFoundException('The API manager is not found');
+        }
+    }
+
+
+    public function perform(): void
+    {
+//        $response = $this->api->read('items', $this->id);
+//        dd($response);
+        $apiUrl = $this->serverUrl->getScheme().'://'.$this->serverUrl->getHost()."/api/items/{$this->id}";
+
+        # Step 0 - create graph and define prefix schema:
+        RdfNamespace::set('schema', 'https://schema.org/');
+        $graph = new Graph(); //dep injection?
+
+        $folder = $this->createTemporaryFolder();
+
+        # Step 1 - get lds dataset
+        $graph->parse($apiUrl, 'jsonld');
+
+        $distributionService = new DistributionService();
+        $distribution = $distributionService->getDistribution($graph);
+
+        $itemSets = $graph->resourcesMatching("^schema:mainEntityOfPage");
+
+        $crawler = new ItemSetCrawler();
+
+        foreach ($itemSets as $itemSet) {
+            $item_set_id = $this->getIdFromPath($itemSet->getUri());
+            $crawler->crawl($item_set_id, $folder, $this->serverUrl);
+        }
+
+        $generatedTripples = glob($folder. "/*.nt");
+
+
+// Name of the output file
+        $output_file = "$folder/merged_file.nt";
+
+// Open the output file for writing
+        $handle = fopen($output_file, "w");
+
+// Loop through the file array and append each file to the output file
+        foreach ($generatedTripples as $file) {
+            // Open the current file for reading
+            $handle2 = fopen($file, "r");
+
+            // Read the contents of the current file and append it to the output file
+            fwrite($handle, fread($handle2, filesize($file)));
+
+            // Close the current file
+            fclose($handle2);
+        }
+
+// Close the output file
+        fclose($handle);
+
+        $graph = new \EasyRdf\Graph();
+        $graph->parseFile($output_file);
+
+        $endFile = OMEKA_PATH . '/files/datadumps/'.$distribution->getFilename();
+        file_put_contents($endFile, $graph->serialise("jsonld"));
+
+        $size = (new \SplFileInfo($endFile))->getSize();
+
+        // todo update distributie
+
+//        /** @var Response $entity */
+//        $entity = $this->api->read('items', $distribution->getId());
+//        /** @var ItemRepresentation $item */
+//        $item = $entity->getContent();
+//        $values = $item->values();
+////        $item-
+////        $this->api->update('items', $distribution->getId());
+//            // determine size and names and update record
+
+    }
+
+
+
+    private function createTemporaryFolder() {
+        $dir = sys_get_temp_dir();
+        $tmp = uniqid('lds_');
+        $path = $dir . '/' . $tmp;
+        mkdir($path);
+        return $path;
+    }
+
+    private function getIdFromPath($uri): int {
+        $path = parse_url($uri, PHP_URL_PATH);
+        $segments = explode('/', $path);
+        $id = end($segments);
+        return (int) $id;
+    }
```

Comment:
Given the POC status of this initial code I understand this code is in this class. But please consider moving it away to a helper service that is injected on this Job. But feel free to implement this at a later stage.